lib-opt/GHCup/OptParse/Compile.hs

```diff
@@ -674,3 +674,6 @@ compile compileCommand settings Dirs{..} runAppState runLogger = do
               VLeft e -> do
                 runLogger $ logError $ T.pack $ prettyHFError e
                 pure $ ExitFailure 9
+ where
+  guessMode = if guessVersion settings then GLaxWithInstalled else GStrict
```

Comment:
Note that some subcommands set this to `GLaxWithInstalled` and some to `GLax`. E.g. we don't want to match on installed versions during `ghcup install ghc`, but we do during `ghcup set ghc`.